app/workers/rapid_ready_for_decision/rrd_processor.rb

```diff
@@ -96,15 +97,33 @@ def icn
       account_record = account
       raise AccountNotFoundError, "for user_uuid: #{form526_submission.user_uuid} or their edipi" unless account_record
 
-      account_record.icn.presence
+      return account_record.icn.presence unless account_record.is_a?(Array)
+
+      # Multiple Account records should have the same ICN
+      icns = account_record.pluck(:icn).uniq.compact
+      return icns.first if icns.size == 1
+
+      raise MultipleIcnsFound, "for the user '#{form526_submission.user_uuid}': #{icns}"
     end
 
     def account
       account = Account.lookup_by_user_uuid(form526_submission.user_uuid)
       return account if account
 
       edipi = form526_submission.auth_headers['va_eauth_dodedipnid'].presence
-      Account.find_by(edipi: edipi) if edipi
+      account_matching_edipi(edipi)
+    end
+
+    # There's no DB constraint that guarantees uniqueness of edipi, so return an array if there are multiple Accounts
+    def account_matching_edipi(edipi)
+      return nil unless edipi
+
+      accounts = Account.where(edipi: edipi).to_a
+      return nil if accounts.empty?
+
+      return accounts.first if accounts.size == 1
+
+      accounts
```

Comment:
I think it's pretty confusing for this method to return either a single account or an array of accounts (here in this method as well as where it's used above).

Reply:
Yea. Any alternative ideas to make this less confusing?

Reply:
Updated to return array of Accounts